packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -570,52 +615,21 @@ class FlutterPlugin implements Plugin<Project> {
         }
     }
 
-    private Properties getPluginList() {
-        File pluginsFile = new File(project.projectDir.parentFile.parentFile, '.flutter-plugins')
-        Properties allPlugins = readPropertiesIfExist(pluginsFile)
-        Properties androidPlugins = new Properties()
-        allPlugins.each { name, path ->
-            if (doesSupportAndroidPlatform(path)) {
-                androidPlugins.setProperty(name, path)
-            }
-            // TODO(amirh): log an error if this plugin was specified to be an Android
-            // plugin according to the new schema, and was missing a build.gradle file.
-            // https://github.com/flutter/flutter/issues/40784
-        }
-        return androidPlugins
+    /** Gets the list of plugins that support the Android platform. */
```

Comment:
Can you add the comment you gave john above about this Object being a List<String>, String or bool to this documentation? 

Secondly can you have this function take in a project object instead of nothing?

Reply:
Or alternatively link to the documentation you have in shared.groovy?

Reply:
In this case I am asking for you to update the 6 call sites. The reason has to do with readability, global variables like project can make code difficult to reason about. Especially the code in the builds system with high impact and a very small number of people who are comfortable working on it we need to optimize on readability or if you prefer understandability. This method is as convenience function which does help with readability but not relying on a global variable will help even more. It will also help as we migrate the bulk of flutter.groovy to smaller files by not needing access to the global project object to move this helper function. 

I hope that description was helpful.

Reply:
I see. So to understand this correctly, the dependency on the project variable is not wanted, but it should rather be handed over to the according methods, as it is first initialized with the `apply` method. 
Should this concept be applied to all methods then?

Done via 75d3ff252f61318bd3c0534149e69e5b96b00a59

Reply:
Most of the methods still use the `project` attribute though.